lib/Sema/TypeOfReference.cpp

```diff
@@ -1514,6 +1514,18 @@ DeclReferenceType ConstraintSystem::getTypeOfMemberReference(
     // Wrap it in a metatype.
     memberTy = MetatypeType::get(memberTy);
 
+    // However, if overload resolution finds a value generic decl from name
+    // lookup, replace the returned member type to be the underlying value type
+    // of the generic.
+    //
+    // This can occur in code that does something like: 'type(of: x).a' where
+    // 'a' is the static value generic member.
```

Comment:
This should be handled in substMemberTypeWithBase() because that's where we end up if you resolve a type alias member this way, etc.

Reply:
I agree, but this function also wraps the result of substMemberTypeWithBase() in a Metatype so it seems we'll still need to have special handling in this function to prevent that in the value generic case.